arangod/Transaction/Context.h

```diff
@@ -141,6 +165,9 @@ class Context {
     TransactionId id;
     bool isReadOnlyTransaction;
     bool isFollowerTransaction;
+    bool isStreamingTransaction = false;
+    bool isEarlyReadOnly = false;
```

Comment:
It is puzzling that there are now 2 different read-only flags for one transaction, which may or may not contradict each other. I think we should go with only a single flag, preferrably `isReadOnlyTransaction`.
The name `isEarlyReadOnly` leaves a lot of questions, mainly, what does "early" mean, if it will be accurate after the "early" stage of the transaction etc. Better to have only a single, reliable flag.